components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/DefaultServiceURLBuilder.java

```diff
@@ -124,7 +125,12 @@ protected String getResolvedUrlPath(String tenantDomain) {
         if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled() && !resolvedUrlContext.startsWith("t/") &&
                 !resolvedUrlContext.startsWith("o/")) {
             if (mandateTenantedPath || isNotSuperTenant(tenantDomain)) {
-                resolvedUrlStringBuilder.append("/t/").append(tenantDomain);
+                if (IdentityCoreServiceDataHolder.getInstance().isOrganizationManagementEnabled() &&
+                        PrivilegedCarbonContext.getThreadLocalCarbonContext().getOrganizationId() != null) {
+                    resolvedUrlStringBuilder.append("/o/").append(tenantDomain);
+                } else {
+                    resolvedUrlStringBuilder.append("/t/").append(tenantDomain);
+                }
             }
```

Comment:
The tenant/org domain append logic is further improved by this commit.
https://github.com/wso2/carbon-identity-framework/pull/4229/commits/2498e8b0a4bd517cdae65f97acbee8b3dfdf2447

1 - organization management enable check put inside the `if (mandateTenantedPath || isNotSuperTenant(tenantDomain)) {` condition.  (This method gets called very frequently and needed to optimize)

2 - Added a logic to check organizationId set in carbon context as it is added only for the requests with `o/<org-domain>` (Help to avoid appending `/o/` for other tenanted resources not related to organization management)